tests/test_expression.py

```diff
@@ -150,7 +150,7 @@ def get_context_data(
 
         self.assertEqual(
             rendered.strip(),
-            "<!-- _RENDERED SimpleComponent_5b8d97,a1bc3e -->\n                <div>lorem</div>\n                <div>True</div>\n                <div>[{'a': 1}, {'a': 2}]</div>",  # noqa: E501
+            "<!-- _RENDERED SimpleComponent_5b8d97,a1bc3f,, -->\n<div data-djc-id-a1bc3f>lorem</div>\n<div data-djc-id-a1bc3f>True</div>\n<div data-djc-id-a1bc3f>[{'a': 1}, {'a': 2}]</div>",  # noqa: E501
```

Comment:
The HTML comment that we leave behind after a component was rendered, e.g. `<!-- _RENDERED SimpleComponent_5b8d97`, now contains also info about which JS and CSS variables should be loaded with the specific rendering of the component.

Reply:
What do the three values mean? Is it just a list of component ids?

Reply:
Fields are:
- `comp_cls_hash` - Cache key of the component class that was rendered
- `id` - Component render ID - Will be used for passing JS data to individual component instances.
- `js` - Cache key for the JS data from `get_js_data()` - part of JS variables feature
- `css` - Cache key for the CSS data from `get_css_data()` - part of CSS variables feature

I added a comment here to make it clear https://github.com/EmilStenstrom/django-components/pull/861/files#diff-371d604ce82d2a01458065f1b137fd981b12af1b0b5c5fb2c1f7f5b45c20003fR403

Reply:
Btw I just realized that, since the "component ID" is now different for each time we call `render()`, then the way we pass the information above could be simplified, such that:

1. We'd keep only the render ID inside the HTML:
    ```html
    <!-- _RENDERED a1bc3e -->
    ```
2. When a component is rendered, we'd cache the other related data (`comp_cls_hash`, `js`, `css`) in an actual cache, instead of inserting it into the HTML:
    ```py
    cache.set(
        "a1bc3e",
        {"comp_cls_hash": "SimpleComponent_5b8d97", "js": None, "css": None},
    )
    ```
3. Then, when the rendered HTML gets processed by the middleware or `render_dependencies()`, it would extract only the render ID from `<!-- _RENDERED a1bc3e -->`, and get (pop) the rest of the data from the cache.

This was not possible before, when multiple calls to `render()` used the same `component_id`, because it wasn't possible to tell how long the entries in the cache would have to remain alive for.

I won't make the changes in this PR, tho, but sometime later.

Reply:
Using the built-in Django cache requires configuration on the user part, so I'm not sure the trade-off is worth it. Let's talk in a separate PR about that :)

Reply:
Ah, by cache here I meant just a simple dictionary :D